File: frontend/src/services/adminOverview.ts
import api from "@/services/api";

export interface AdminInboxCounts {
  pending_invites: number;
  pending_companies: number;
  pending_jobs: number;
  new_applications: number;
  oldest_pending_company_days: number | null;
  oldest_pending_job_days: number | null;
  oldest_new_application_days: number | null;
}

export interface TopJobEntry {
  id: number;
  title: string;
  application_count: number;
}

export interface AdminStatsCounts {
  active_companies: number;
  published_jobs: number;
  total_candidates: number;
  application_status_counts: Record<string, number>;
  top_jobs: TopJobEntry[];
}

export interface RecentItem {
  type: "company" | "job" | "application";
  label: string;
  sublabel: string | null;
  created_at: string;
}

export interface TrendPoint {
  date: string;
  n: number;
}

export interface AdminPulse {
  new_candidates_7d: number;
  new_applications_7d: number;
  recent_items: RecentItem[];
  trend_30d: TrendPoint[];
}

export interface AdminOverviewRead {
  inbox: AdminInboxCounts;
  stats: AdminStatsCounts;
  pulse: AdminPulse;
}

export async function getAdminOverview(signal?: AbortSignal): Promise<AdminOverviewRead> {
  const res = await api.get<AdminOverviewRead>("/api/admin/overview", { signal });
  return res.data;
}

File: libs/shared/rs_shared/schemas/admin_overview.py
"""Admin overview response schema."""

from pydantic import BaseModel


class TopJobEntry(BaseModel):
    id: int
    title: str
    application_count: int


class RecentItem(BaseModel):
    type: str  # "company" | "job" | "application"
    label: str
    sublabel: str | None
    created_at: str  # ISO 8601


class AdminInboxCounts(BaseModel):
    pending_invites: int
    pending_companies: int
    pending_jobs: int
    new_applications: int
    oldest_pending_company_days: int | None
    oldest_pending_job_days: int | None
    oldest_new_application_days: int | None


class AdminStatsCounts(BaseModel):
    active_companies: int
    published_jobs: int
    total_candidates: int
    application_status_counts: dict[str, int]
    top_jobs: list[TopJobEntry]


class TrendPoint(BaseModel):
    date: str  # ISO 8601 date (YYYY-MM-DD)
    n: int


class AdminPulse(BaseModel):
    new_candidates_7d: int
    new_applications_7d: int
    recent_items: list[RecentItem]
    trend_30d: list[TrendPoint]


class AdminOverviewRead(BaseModel):
    inbox: AdminInboxCounts
    stats: AdminStatsCounts
    pulse: AdminPulse
